src/vm/mod.rs

```diff
@@ -1,3 +1,80 @@
 // SPDX-License-Identifier: Apache-2.0
 
 mod linux;
+
+use crate::vm::linux::{
+    ioctl::Cmd,
+    types::{Capabilities, CpuidConfig},
+};
+use kvm_ioctls::{Error, Kvm, VmFd};
+
+// Defined in linux/arch/x86/include/uapi/asm/kvm.h
+const KVM_X86_TDX_VM: u64 = 2;
+
+/// Handle to the KVM VM file descriptor
+pub struct TdxVm(VmFd);
+
+impl TdxVm {
+    /// Create a new TDX VM with KVM
+    pub fn create(kvm_fd: Kvm) -> Result<Self, Error> {
+        let vm_fd = kvm_fd.create_vm_with_type(KVM_X86_TDX_VM)?;
+        Ok(TdxVm(vm_fd))
+    }
+
+    /// Retrieve information about the Intel TDX module
+    pub fn get_capabilities(&self) -> Result<TdxCapabilities, kvm_ioctls::Error> {
+        // configure the kernel facing capabilities structure
+        let caps = Capabilities::new();
+
+        // configure the kernel facing TDX command that will be wrapped by the
+        // MEMORY_ENCRYPT_OP ioctl
+        let mut cmd: Cmd = Cmd::from(&caps);
+
+        unsafe { self.0.encrypt_op(&mut cmd)? };
+
+        // MEMORY_ENCRYPT_OP ioctl was a success so we can return the TDX
+        // capabilities in a user-friendly way
+        Ok(TdxCapabilities {
+            fixed_attrs: FixedAttrs {
+                fixed0: caps.attrs_fixed0,
+                fixed1: caps.attrs_fixed1,
+            },
+            fixed_xfam: FixedXFAM {
+                fixed0: caps.xfam_fixed0,
+                fixed1: caps.xfam_fixed1,
+            },
+            supported_gpaw: caps.supported_gpaw,
+            cpuid_configs: Vec::from(caps.cpuid_configs),
+        })
+    }
+}
+
+/// Reflects the Intel TDX module capabilities and configuration and CPU
+/// capabilities
+#[derive(Debug)]
+pub struct FixedAttrs {
+    /// If any bit is a 0 in the bitmap, it must be 0 in a TD's ATTRIBUTES
```

Comment:
What bitmap are you referring to here?

Reply:
See above comment. Will add better documentation.